onnxruntime/core/session/inference_session.cc

```diff
@@ -956,21 +956,12 @@ class InferenceSession::Impl {
                                  const std::vector<std::string>& custom_list) {
     auto add_transformers = [&](TransformerLevel level, std::vector<std::string>&& providers, std::string t_name) {
       // Generate and register rewrite rules for level
-      auto rewrite_rules_to_register =
-          transformer_utils::GenerateRewriteRules(level, &custom_list);
-      if (!rewrite_rules_to_register.empty()) {
-        std::unique_ptr<RuleBasedGraphTransformer> graph_rewrite_rules =
-            std::make_unique<TopDownRuleBasedTransformer>(t_name + "_RuleBasedTransformer",
-                                                          "Apply rewrite rules for " + t_name);
-        for (auto& entry : rewrite_rules_to_register) {
-          graph_rewrite_rules->Register(std::move(entry));
-        }
-        transformer_manager.Register(std::move(graph_rewrite_rules), level,
-                                     std::move(providers));
-      }
+      std::unique_ptr<RuleBasedGraphTransformer> rule_transformer =
```

Comment:
>rule_transformer [](start = 49, length = 16)

Since you are moving the rule based transformer generation to utils... I am wondering can we also move this detail to utils under GenerateTransformers?

So basically inference_session can just call GenerateTransfomers and not care whether some are rulebased vs some are stand alone

Reply:
Yes, exactly, that's what I would prefer too. I will take care of it.